doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2375 @@
+
+##### 4.2.3.2 Flag Change Count and Time Set/Clear Behavior During Transceiver Removal and Insertion
+
+When a transceiver is removed, `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables are deleted by the `SfpStateUpdateTask` thread.
+
+When the transceiver is inserted back, the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables are recreated through the periodic polling routine of `DomInfoUpdateTask` and the flag change count and set/clear time are updated based on the current flag status.
+
+#### 4.2.4 Diagnostic Information Last Update Timestamp and Interval Period by `DomInfoUpdateTask`
+
+All the diagnostic tables (except for the metadata tables storing change count and last set/clear time) contain the `table_last_update_time` field to capture the last update timestamp.
+Specifically, the `TRANSCEIVER_STATUS` table contains the `diagnostics_update_interval` field to capture the interval period at which the diagnostic information is updated by the `DomInfoUpdateTask` thread for a port. This field is not present in the other diagnostic tables since the diagnostic information is updated for all ports in a sequential manner.
+
+1. **`table_last_update_time`**:
+   - This field records the timestamp (in local timezone) at which the corresponding diagnostic information was last updated by the `DomInfoUpdateTask` thread for a port.
+   - The timestamp is recorded in the format `Day Mon DD HH:MM:SS YYYY`.
+
+2. **`diagnostics_update_interval`**:
+   - This field records the interval period (in seconds) at which the diagnostic information is updated by the `DomInfoUpdateTask` thread for a port.
+   - Unlike the `dom_info_update_periodic_secs` value, this field is updated dynamically to reflect the actual time taken to update the diagnostic information for a port in the `redis-db`.
+   - The time taken to read the diagnostic information from the transceiver can vary between successive polls based on:
+     - Transceiver type
+     - Number of diagnostic parameters supported by the transceiver
+     - Device platform (which can affect I2C read/write speed)
+     - Number of ports with transceivers plugged in
+     - Number of ports in the breakout port group
+     - Number of link change events between successive polls
+
+#### Calculation of `diagnostics_update_interval` Using EWMA
+
+The Exponentially Weighted Moving Average (EWMA) is used to calculate the `diagnostics_update_interval`, providing a smooth and responsive average of the update intervals. An alpha value of 0.1 is used for the EWMA calculation to give more weight to historical intervals.
+
+**Formula:**
+
+$$
```

Comment:
@mihirpat1 can you check the rendered text output of the formula? seems not readable